### DISCUSSION TITLE
GitExtensions Questions

### DISCUSSION THREAD

**Discussion by O-R-J** [gitextensions/gitextensions#10000] [Q&A] [answered]
Hello,

I am currently conducting security assessments on behalf of Hitachi Energy, and one of the applications I have ben asked to assess is GitExtensions. 
Could anyone please answer the following questions for me?

- Is version 3.5.4 supported? (will receive security updates, hotfixes or otherwise)

- Does version 3.5.4 have any known identified vulnerabilities?

- Are there any plans to discontinue this software? (EOL)

- Does this application collect app logs?

Thank you in advance for your time!

Kind regards,
Oliver Jenkins

**Comment by gerhardol** [comment, marked as answer]
>     * Is version 3.5.4 supported? (will receive security updates, hotfixes or otherwise)

Supported if something urgent comes up, no release planned.

>     * Does version 3.5.4 have any known identified vulnerabilities?

No known. There are normally very few in applications like GE, no server functionality and few direct accesses.

There may be in the bundled PuTTY binary. (That is optional to use, recommendation is to use an external installation).
Similar, before 3.0 Git was bundled, those versions had occasionally security issues (external Git could be used there too.)

Note: PuTTY is no longer bundled in next GE version.

>     * Are there any plans to discontinue this software? (EOL)

No

>     * Does this application collect app logs?

You can enable a Git Command Log, but you have to save the log manually.
Telemetry may be enabled to supply some anonymous usage data for the app.

**Reply by RussKie** [reply]
Here's what the app collects for the telemetry purposes: https://github.com/gitextensions/gitextensions/blob/master/PrivacyPolicy.md

**Comment by O-R-J** [comment]
Thank you for the information so far Gerhard,

Regarding logs, you have mentioned that the logs need to be saved manually. How often do these logs need to be saved in order to prevent the loss of these logs?


Kind regards,
Oliver Jenkins

**Reply by pmiossec** [reply]
> Regarding logs, you have mentioned that the logs need to be saved manually. How often do these logs need to be saved in order to prevent the loss of these logs?

As a normal user, you don't care about these logs. You display it only when you want to investigate a behavior that you don't understand or a bug. And you only save it if you want to communicate the behavior or the issue with other users or the GitExtensions development team.

Your question was about collecting logs that's why @gerhardol insisted on the manually process.
He is talking about these logs that are kept in memory:
![image](https://user-images.githubusercontent.com/460196/169539483-42b6f803-d905-43ae-abc1-74d1e07ed7b2.png)



GitExtensions don't produce nor send logs that GitExtensions team can collect.

**Comment by O-R-J** [comment]
Thank you Phillippe,

Could you please tell me whether these are error logs that you are talking about here?

I assume a bug would result in an error code, but would it be logging the error in the code that has been developed, or the error in the application itself?


Kind regards,
Oliver Jenkins

**Reply by vbjay** [reply]
Nope.  Gitextensions wraps git.  So what did GE do?  The log is how you find that out.   I cleared the log and hit refresh so you can see the commands it ran in the refresh.  
[log.txt](https://github.com/gitextensions/gitextensions/files/8755127/log.txt)

**Reply by vbjay** [reply]
Saved by doing this 

![image](https://user-images.githubusercontent.com/3629489/169835661-59140997-cd80-427f-9742-d3f0fadcdfc1.png)

**Reply by vbjay** [reply]
The logs are not an audit trail of the app forever.  They are there so the developer can look at the commands and see what doing x in the app does to the repo.  They are diagnostic use for the developer only and for the maintainers in this project.  Someone can clear the log and then perform x action and send logs and that can be useful in figuring out what caused x problem, if they reported an issue after doing something in the app.
